src/main/java/green/citibike/CitiController.java

```diff
@@ -0,0 +1,140 @@
+package green.citibike;
+
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import green.citibike.json.StationInfo;
+import green.citibike.json.StationStatus;
+import green.citibike.json.Stations;
+import green.citibike.json.StatusInfo;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class CitiController {
+    private final Stations<StationInfo> stationsInfo;
+    private final Stations<StatusInfo> statusInfo;
+    HashMap<String, StationStatus> stationStatusMap = new HashMap<>();
+    List<StationStatus> stationStatusList;
+    final double RADIUS = 3958.8;
+
+    public CitiController(Stations<StationInfo> stationsInfo, Stations<StatusInfo> statusInfo) {
+        this.stationsInfo = stationsInfo;
+        this.statusInfo = statusInfo;
+        mergeStationStatus();
+        stationStatusList = stationStatusMap.values().stream().toList();
+    }
+
+    private void mergeStationStatus() {
+        List<StationInfo> stationInfoList = stationsInfo.data.stations;
+        List<StatusInfo> statusInfoList = statusInfo.data.stations;
+
+        insertStationInfo(stationInfoList);
+        insertStatusInfo(statusInfoList);
+    }
+
+    private void insertStationInfo(List<StationInfo> stationInfoList) {
+        for (StationInfo stationInfo : stationInfoList) {
+            stationStatusMap.put(stationInfo.station_id, new StationStatus(stationInfo.lon,
+                    stationInfo.lat, stationInfo.name, stationInfo.station_id));
+        }
+    }
+
+    private void insertStatusInfo(List<StatusInfo> statusInfoList) {
+        for (StatusInfo statusInfo : statusInfoList) {
+            StationStatus staSta = stationStatusMap.get(statusInfo.station_id);
+            staSta.addStatus(statusInfo.isRenting,
+                    statusInfo.num_bikes_available, statusInfo.num_docks_available);
+            stationStatusMap.replace(statusInfo.station_id, staSta);
+        }
+    }
+
+    public List<StationStatus> getStationStatus() {
+        return stationStatusList;
+    }
+
+    public StationStatus getStatusInfo(String stationId) {
+        return stationStatusMap.get(stationId);
+    }
+
+    private String findClosestStationCoords(double lat, double lon) {
+        double closest = Double.MAX_VALUE;
+        String closestStationId = stationStatusList.get(0).station_id;
+
+        for (StationStatus stationStatus : stationStatusList) {
+            double distance = haversine(stationStatus.lat, stationStatus.lon, lat, lon);
+            if (distance < closest) {
+                closest = distance;
+                closestStationId = stationStatus.station_id;
+            }
+        }
+
+        return closestStationId;
+    }
+
+    //reference https://www.geeksforgeeks.org/haversine-formula-to-find-distance-between-two-points-on-a-sphere/
+    private double haversine(double lat1, double lon1, double lat2, double lon2) {
```

Comment:
Regular distance formula will suffice for this. You can keep this but it wasn't necessary.